processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -39,14 +48,29 @@ public SegmentMetadataQuery(
       @JsonProperty("intervals") QuerySegmentSpec querySegmentSpec,
       @JsonProperty("toInclude") ColumnIncluderator toInclude,
       @JsonProperty("merge") Boolean merge,
-      @JsonProperty("context") Map<String, Object> context
+      @JsonProperty("context") Map<String, Object> context,
+      @JsonProperty("usingDefaultInterval") Boolean useDefaultInterval
```

Comment:
The flag needs to be in the constructor, in the situations where a query object is generated from an existing object, e.g. query.withOverriddenContext, query.withQuerySegmentSpec, the querySegmentSpec will not be null.